android/app/src/main/cpp/app/organicmaps/bookmarks/data/BookmarkManager.cpp

```diff
@@ -789,6 +789,32 @@ Java_app_organicmaps_bookmarks_data_BookmarkManager_nativeSetBookmarkParams(
   g_framework->ReplaceBookmark(static_cast<kml::MarkId>(bmk), bmData);
 }
 
+JNIEXPORT void JNICALL
+Java_app_organicmaps_bookmarks_data_BookmarkManager_nativeSetTrackParams(
+    JNIEnv * env, jclass, jlong trackId,
+    jstring name, jint color, jstring descr)
+{
+  auto const * nTrack = frm()->GetBookmarkManager().GetTrack(static_cast<kml::TrackId>(trackId));
+  ASSERT(nTrack, ("Track must not be null with id:", trackId));
+
+  kml::TrackData trackData(nTrack->GetData());
+  auto const trkName = jni::ToNativeString(env, name);
+  if (nTrack->GetName() != trkName)
+    kml::SetDefaultStr(trackData.m_name, trkName);
+  if (descr)
+    kml::SetDefaultStr(trackData.m_description, jni::ToNativeString(env, descr));
+  trackData.m_layers[0].m_color.m_rgba = static_cast<uint32_t>((color << 8) + 255);
```

Comment:
Actually android sends color to core in ARGB integer format in which first 8 bits of integer are for opacity and its represented by A in (A)RGB, but in core colors are used in RGBA format. Since opacity for all colors are always 100% i removed 8 bits from start and shifted all bits to left by 8 bits and then added 8 ones in back assuming 100% opacity.

here opacity is not preserved, the better way would be to take out those 8 bits from front and insert them at the end so that opacity stay preserved.

I can create a separate method ARGBtoRGBAconverter() and use it.